src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs

```diff
@@ -36,6 +36,18 @@ public class AnonymousBodyParameter : TypedRule<SwaggerParameter>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
+
+        /// <summary>
+        /// When to apply the validation rule, before or after it has been merged as a part of 
+        /// its merged document as specified in the corresponding '.md' file
+        /// By default consider all rules to be applied for After only
+        /// </summary>
+        public override ServiceDefinitionDocumentState ValidationRuleMergeState => ServiceDefinitionDocumentState.Individual;
```

Comment:
I think I asked a question similar to Vishrut in a prior PR, would it be helpful to define a "default" for the cases where it doesn't matter, so we can refer to it when that's the case, and explicitly call it out when it does?